server/sonar-server/src/main/java/org/sonar/server/component/ws/ProjectMeasuresQueryValidator.java

```diff
@@ -40,7 +44,8 @@ public ProjectMeasuresQueryValidator(DbClient dbClient) {
   }
 
   public void validate(DbSession dbSession, ProjectMeasuresQuery query) {
-    Set<String> metricKeys = query.getMetricCriteria().stream().map(MetricCriterion::getMetricKey).collect(Collectors.toSet());
+    Set<String> metricKeys = new HashSet<>(query.getMetricCriteria().stream().map(MetricCriterion::getMetricKey).collect(Collectors.toSet()));
+    Optional.ofNullable(query.getSort()).filter(sort -> !SORT_BY_NAME.equals(sort)).ifPresent(metricKeys::add);
```

Comment:
what you want to do: you want to add to metric keys the sort if it's not a name. I think the code doesn't reflect your intent. Let's discuss